azext_iot/common/certops.py

```diff
@@ -125,3 +134,201 @@ def open_certificate(certificate_path: str) -> str:
         raise ValueError("Certificate file type must be either '.pem' or '.cer'.")
     # Remove trailing white space from the certificate content
     return certificate.rstrip()
+
+
+def create_root_certificate(
+    subject: str = "Azure_IoT_CLI_Extension_Cert",
+    valid_days: int = 365,
+    key_size: int = 4096,
+) -> Dict[str, str]:
```

Comment:
do we want to be able to create certs with passwords? (this applies to the function after this too)

Reply:
I don't think so. I think that's best left to the users to create their own certs / keys with passwords, and then they can BYOK with the `--root-cert`, `--root-key`, and `root-pass` parameters.

I think it's a possibility and a nice-to-have, but the way the logic works now, we'd end up overloading `--root-pass` with a "set root password" if creating a self-signed cert, or "use root password" if using an existing protected key.